### PR TITLE
Add Syncer worker

### PR DESCRIPTION
**Pull request by danielrozenberg**
This PR adds a second worker to this repo, which will be used as a GitHub App that syncs configurations from the `ampproject/amphtml` repo to the CDN worker's KV storage - specifically it can sync the client-side experiments file (AMP_EXP config) and versioning (RTV) file. I make some assumptions on the location and structure of the versioning file, but those assumptions are easy to change once we decide about them.

For now there are many duplicate files between the two directories, I figured we can dedup them later as part of a larger infrastructure PR

### CODE REVIEW

**Review by rsimha — APPROVED** [review]

**Review by rileyajones — APPROVED** [review]

File: syncer/src/amp_exp.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    console.log(AMP_EXP_FILE, 'fails validation:', error);
+    octokit.rest.issues.createComment({
+      owner,
+      repo,
+      'issue_number': pullRequest.number,
+      body:
+        `This pull requested modified the [client-side experiments config file](https://github.com/${owner}/${repo}/blob/${sha}/${AMP_EXP_FILE}) but it does not pass schema validation.\n\n` +
+        `Please send a follow-up pull request to fix this file using the [correct schema](https://github.com/${owner}/${repo}/blob/${sha}/build-system/global-configs/README.md#client-side-experiments-configjson).`,
+    });
+    return;
```
Comment (estherkim) [inline syncer/src/amp_exp.ts:68]: how about `await octokit.rest.issues.createComment({...});`

File: syncer/src/handler.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    const {data} = await app.octokit.request('GET /app');
+    return new Response(JSON.stringify(data));
+  }
+
+  const id = request.headers.get('X-Github-Delivery');
+  const name = request.headers.get('X-Github-Event') as WebhookEventName;
+  const signature = request.headers
+    .get('X-Hub-Signature-256')
+    ?.slice('sha256='.length);
+  const payload = await request.json();
```
Comment (estherkim) [inline syncer/src/handler.ts:32]: nit: do the null checks below before this `await request.json()` for the tiniest of time saved

File: syncer/src/amp_exp.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
+/**
+ * Client-side experiments (AMP_EXP) sync code.
+ */
```
Comment (estherkim) [inline syncer/src/amp_exp.ts:3]: Can this comment fit on one line? Same suggestion for all files.

File: syncer/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * Contains request routing logic.
+ */
+
+import {App} from '@octokit/app';
+import {Octokit} from '@octokit/rest';
+
+import {handleRequest} from './handler';
+import {pullRequestClosedHandler} from './webhook';
+
+// Secrets defined by `wrangler secret`.
```
Comment (estherkim) [inline syncer/src/index.ts:11]: Why are we using wrangler to manage secrets instead of a gh actions environment? To be honest I don't know what wrangler is.

File: syncer/test/.eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (estherkim) [inline syncer/test/.eslintrc.json:1]: consolidate this config with `syncer/` or better yet, the to-be root config... there's too many configs!

File: syncer/test/fixtures/webhook.pull_request.closed.json.ts [hunk 6]
```diff
@@ -0,0 +1,678 @@
+import {PullRequestClosedEvent} from '@octokit/webhooks-types';
+
+export default {
```
Comment (estherkim) [inline syncer/test/fixtures/webhook.pull_request.closed.json.ts:3]: do you really need all this data for tests, or can you create an object with only the data you need?

File: syncer/test/webhook.test.ts [hunk 7]
```diff
@@ -0,0 +1,165 @@
+/**
+ * Tests for webhook.ts.
+ */
```
[the remark this answers is not in the document]
Comment (danielrozenberg) [reply syncer/test/webhook.test.ts:3]: Done. I thought for some reason that this was required by the ESLint config but it passes the lint checks without it 😅

File: syncer/src/amp_exp.ts [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+export const AMP_EXP_FILE =
+  'build-system/global-configs/client-side-experiments-config.json';
+
+/**
+ * Sync AMP_EXP config.
+ *
+ * @param sha -
+ * @param octokit -
+ * @param payload -
+ */
```
[the remark this answers is not in the document]
Comment (danielrozenberg) [reply syncer/src/amp_exp.ts:27]: I was gonna fill these out, but they're pretty self explanatory so I didn't bother, and then forgot to shorten this to a one-liner function doc. Done.

File: syncer/src/amp_exp.ts [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+      body: `This pull request's changes to the [client-side experiments config file](https://github.com/${owner}/${repo}/blob/${sha}/${AMP_EXP_FILE}) have been deployed to the CDN`,
+    });
+  } catch (error) {
+    console.log(AMP_EXP_FILE, 'fails validation:', error);
+    octokit.rest.issues.createComment({
+      owner,
+      repo,
+      'issue_number': pullRequest.number,
+      body:
+        `This pull requested modified the [client-side experiments config file](https://github.com/${owner}/${repo}/blob/${sha}/${AMP_EXP_FILE}) but it does not pass schema validation.\n\n` +
```
[the remark this answers is not in the document]
Comment (danielrozenberg) [reply syncer/src/amp_exp.ts:65]: Done

File: syncer/src/index.ts (hunk 4 above)
Comment (danielrozenberg) [reply syncer/src/index.ts:11]: This raises an important enough question to elevate this to the main discussion, see below...
